### PR TITLE
[rllib] Document "v2" APIs

### PR DESCRIPTION
**Pull request by ericl**
## What do these changes do?

This is a pretty comprehensive re-org of the docs to cover the new API features. I also took the chance to clean up the RLlib directory structure, adding `env`, `agents`, and `evaluation` sub-dirs.

The only section left to write is "RL building blocks", which I'm going to leave for another PR.

### CODE REVIEW

File: python/ray/rllib/README.rst [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-- Deep Q Networks (`DQN <https://github.com/ray-project/ray/tree/master/python/ray/rllib/dqn>`__).
-
-- Deep Deterministic Policy Gradients (`DDPG <https://github.com/ray-project/ray/tree/master/python/ray/rllib/ddpg>`__).
-
-- Ape-X Distributed Prioritized Experience Replay, including both `DQN <https://github.com/ray-project/ray/blob/master/python/ray/rllib/dqn/apex.py>`__ and `DDPG <https://github.com/ray-project/ray/blob/master/python/ray/rllib/ddpg/apex.py>`__ variants.
-
-- Evolution Strategies (`ES <https://github.com/ray-project/ray/tree/master/python/ray/rllib/es>`__), as described in `this paper <https://arxiv.org/abs/1703.03864>`__.
-
-These algorithms can be run on any OpenAI Gym MDP, including custom ones written and registered by the user.
+RLlib: Scalable Reinforcement Learning
```
Comment (richardliaw) [inline python/ray/rllib/README.rst]: Let's add the logo on this page.

File: doc/source/rllib.rst [hunk 2]
```diff
@@ -1,347 +1,73 @@
...
-
-  pip install 'ray[rllib]'
+  pip install ray[rllib]
 
 You might also want to clone the Ray repo for convenient access to RLlib helper scripts:
 
 .. code-block:: bash
 
   git clone https://github.com/ray-project/ray
+  cd ray/python/ray/rllib
```
Comment (richardliaw) [inline doc/source/rllib.rst]: we should just push these helper functions into CLI; it's way cleaner than having users clone the repository.

One dumb suggestion:
```

File: doc/source/rllib.rst [hunk 3]
```diff
@@ -1,347 +1,73 @@
...
 
+Training APIs
```
Comment (richardliaw) [inline doc/source/rllib.rst]: We should have a quick start on this page (see https://gym.openai.com/, https://github.com/uber/horovod, etc).

Comment (richardliaw) [reply doc/source/rllib.rst]: And likely split into 3 use cases: running a pre-made algorithm (1 line)

**Review by richardliaw — APPROVED** [review]
let's put in the logo and then merge first.

File: doc/source/rllib.rst (hunk 2 above)
Comment (ericl) [reply doc/source/rllib.rst]: Agree

File: doc/source/rllib-package-ref.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+----------------
+
+.. automodule:: ray.rllib.models
+    :members:
+
+ray.rllib.optimizers
+--------------------
+
+.. automodule:: ray.rllib.optimizers
+    :members:
```
[the remark this answers is not in the document]
Comment (ericl) [reply doc/source/rllib-package-ref.rst]: Hmm any idea how, it's a dict?

File: doc/source/rllib.rst [hunk 5]
```diff
@@ -1,347 +1,73 @@
-Ray RLlib: Scalable Reinforcement Learning
-==========================================
+RLlib: Scalable Reinforcement Learning
```
[the remark this answers is not in the document]
Comment (ericl) [reply doc/source/rllib.rst]: Hm couldn't figure out how to float it on the right -- a full width rllib is a bit too much.

File: doc/source/rllib-package-ref.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/rllib-package-ref.rst]: hm I think you just need to show `_init` for each member, right?

Something along the lines of https://stackoverflow.com/questions/15903577/include-specific-special-methods-in-sphinx

and https://github.com/ray-project/ray/blame/d75b39f6dfd615fb0a6b47ed3177d96c1fe1fa04/doc/source/tune.rst#L247

File: doc/source/rllib.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (richardliaw) [reply doc/source/rllib.rst]: I can take a shot later

**Comment by richardliaw** [comment]
tests need to be updated after classes were moved around

**Pull request merged**